maven_iuvs/echelle.py

```diff
@@ -0,0 +1,1292 @@
+    spabinlo, spabinhi, nspapix = pix_to_bin(hdul,
+                                             spalo, spahi, 'SPA')
+    spelo, spehi = spepixrange
+    spebinlo, spebinhi, nspepix = pix_to_bin(hdul, 
+                                             spelo, spehi, 'SPE')
+
+    npix = nspapix*nspepix
+
+    if binning['nspa'] == 0 or binning['nspe'] == 0:
+        # data is bad and contains no frames
+        countsperpix = np.nan
+    elif n_int == 1:
+        # single integration
+        countsperpix = np.sum(hdul['Primary'].data[spabinlo:spabinhi, spebinlo:spebinhi])/npix
+    else: # n_int > 1
+        countsperpix = np.sum(hdul['Primary'].data[:, spabinlo:spabinhi, spebinlo:spebinhi], axis=(1,2))/npix    
+        
+    countrate = np.atleast_1d(countsperpix)/hdul['Primary'].header['INT_TIME']
+    
+    if return_npix:
+        return countrate, npix
+    
+    return countrate
+
+
+def get_countrate_diagnostics(hdul):
+    """
+    ...description...
+
+    Parameters
+    ----------
+    hdul : astropy FITS HDUList object
+           HDU list for a given observation
+    spapixrange : 
+    spapixrange : 
+    return_npix : 
+
+    Returns
+    -------
+    countrate : 
+    npix : 
+
+    """
+    Hlya_spapixrange = np.array([slit_start, slit_end])
+    Hlya_countrate, Hlya_npix = get_avg_pixel_count_rate(hdul, Hlya_spapixrange, [450, 505])
+    
+    Hbkg_spapixrange = Hlya_spapixrange + 2*(slit_end-slit_start)
+    Hbkg_countrate, Hbkg_npix = get_avg_pixel_count_rate(hdul, Hbkg_spapixrange, [450, 505])
+    
+    Dlya_countrate, Dlya_npix = get_avg_pixel_count_rate(hdul, Hlya_spapixrange, [415, 450])
+    Dbkg_countrate, Dbkg_npix = get_avg_pixel_count_rate(hdul, Hbkg_spapixrange, [505, 540])
+    
+    return {'Hlya_countrate':Hlya_countrate,
+            'Hlya_npix':Hlya_npix,
+            'Hbkg_countrate':Hbkg_countrate,
+            'Hbkg_npix':Hbkg_npix,
+            'Dlya_countrate':Dlya_countrate,
+            'Dlya_npix':Dlya_npix,
+            'Dbkg_countrate':Dbkg_countrate,
+            'Dbkg_npix':Dbkg_npix}
+
+
+def get_lya_countrates(idx_entry):
+    """
+    Gets Ly α countrates
+    
+    Parameters
+    ----------
+    idx_entry : string
+               folder containing observations, sorted into subfolders labeled by orbit
+
+    Returns
+    -------
+    None -- just updates the index files 
+    """
+    rates = idx_entry['countrate_diagnostics']
+    
+    return {'Hlya':np.nanmean(rates['Hlya_countrate']), 'Hbkg':np.nanmean(rates['Hbkg_countrate']),
+            'Dlya':np.nanmean(rates['Dlya_countrate']), 'Dbkg':np.nanmean(rates['Dbkg_countrate'])}
+
+
+# Metadata -------------------------------------------------------------
+
+
+def get_dir_metadata(the_dir, new_files_limit=None):
+    """
+    Gets metadata for given set of files
+
+    Parameters
+    ----------
+    the_dir : string
+              path to directory containing observation data
+    new_files_limit : 
```

Comment:
number of new files to add before stopping and saving the resulting index.

Reply:
Fixed by d81b483549da109ff45b25607960cdaf97a7345b